service/history/timerQueueTaskExecutorBase.go

```diff
@@ -122,3 +137,14 @@ func (t *timerQueueTaskExecutorBase) getNamespaceIDAndWorkflowExecution(
 		RunId:      task.GetRunID(),
 	}
 }
+
+func (t *timerQueueTaskExecutorBase) deleteHistoryBranch(branchToken []byte) error {
+	if len(branchToken) > 0 {
+		// only delete the history if mutable state is gone
```

Comment:
this comment is out of context here.
here you just delete the given branch, or does it actually check mutable state again?